kubeconfig.go

```diff
@@ -0,0 +1,101 @@
+package kubeconfig
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Config represents the configuration used to create a new kubeconfig library instance.
+type Config struct {
+	micrologger.Logger
+	rest.Config
+}
+
+// TenantCluster provides functionality for connecting to tenant clusters.
+type KubeConfig struct {
+	logger     micrologger.Logger
+	restConfig rest.Config
+}
+
+// New creates a new tenant cluster service.
+func New(config Config) (*KubeConfig, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Config == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Config must not be empty", config)
+	}
+
```

Comment:
For sanity it would be good to copy the config before using it: https://godoc.org/k8s.io/client-go/rest#CopyConfig.

```suggestion
	restConfig := rest.CopyConfig(config.RestConfig)
```